templates/distribution/scripts/pre-apply.sh.tpl

```diff
@@ -190,6 +275,139 @@ deleteTracingMinioHA
+  fi
+{{- end }}
+
+< delete-velero.yaml $yqbin 'select(.kind != "CustomResourceDefinition")' | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+< delete-velero.yaml $yqbin 'select(.kind == "CustomResourceDefinition")' | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+  echo "Velero resources deleted"
+
+}
+
+{{- if eq .reducers.distributionModulesDRType.to "none" }}
+
+{{- if eq .reducers.distributionModulesDRType.from "on-premises" }}
+deleteVelero
+{{- end }}
+
+{{- if eq .reducers.distributionModulesDRType.from "eks" }}
+deleteVelero
+{{- end }}
+
+{{- end }}
+
+{{- end }} # end distributionModulesDRType
+
+# ██    ██ ███████ ██      ███████ ██████   ██████      ██████   █████   ██████ ██   ██ ███████ ███    ██ ██████  
+# ██    ██ ██      ██      ██      ██   ██ ██    ██     ██   ██ ██   ██ ██      ██  ██  ██      ████   ██ ██   ██ 
+# ██    ██ █████   ██      █████   ██████  ██    ██     ██████  ███████ ██      █████   █████   ██ ██  ██ ██   ██ 
+#  ██  ██  ██      ██      ██      ██   ██ ██    ██     ██   ██ ██   ██ ██      ██  ██  ██      ██  ██ ██ ██   ██ 
+#   ████   ███████ ███████ ███████ ██   ██  ██████      ██████  ██   ██  ██████ ██   ██ ███████ ██   ████ ██████  
+                                                                                                                
+
+{{- if index .reducers "distributionModulesDRVeleroBackend" }}
+
+deleteVeleroMinio() {
+
+  $kustomizebin build $vendorPath/modules/dr/katalog/velero/velero-on-prem/minio > delete-dr-minio.yaml
+
+{{- if eq .spec.distribution.modules.monitoring.type "none" }}
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-dr-minio.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-dr-minio-filtered.yaml
+    cp delete-dr-minio-filtered.yaml delete-dr-minio.yaml 
+  fi
+{{- end }}
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-dr-minio.yaml 
+  echo "Minio on kube-system namespace deleted"
+}
+
+{{- if eq .reducers.distributionModulesDRVeleroBackend.to "externalEndpoint" }}
+
+{{- if eq .reducers.distributionModulesDRVeleroBackend.from "minio" }}
+deleteVeleroMinio
+{{- end }}
+
+{{- end }}
+
+{{- end }} # end distributionModulesDRVeleroBackend
+
+# ███    ██  ██████  ██ ███    ██ ██   ██     ████████ ██    ██ ██████  ███████ 
+# ████   ██ ██       ██ ████   ██  ██ ██         ██     ██  ██  ██   ██ ██      
+# ██ ██  ██ ██   ███ ██ ██ ██  ██   ███          ██      ████   ██████  █████   
+# ██  ██ ██ ██    ██ ██ ██  ██ ██  ██ ██         ██       ██    ██      ██      
+# ██   ████  ██████  ██ ██   ████ ██   ██        ██       ██    ██      ███████ 
+
+{{- if index .reducers "distributionModulesIngressNginxType" }}
+
+deleteNginx() {
+
+  $kustomizebin build $vendorPath/modules/ingress/katalog/nginx > delete-nginx.yaml
+  $kustomizebin build $vendorPath/modules/ingress/katalog/dual-nginx > delete-dual-nginx.yaml
```

Comment:
shouldn't we delete single or dual depending on what is installed? I believe we can get that from the `from` field.

Reply:
Yes, but since we are using the --ignore-not-found it's easier to delete them both without doing another condition